tests/test_sample.py

```diff
@@ -151,3 +151,173 @@ def test_sample_exception_wrong_n(self, device, shots):
+        dev.apply("RX", wires=[1], par=[phi])
+        dev.apply("RX", wires=[2], par=[varphi])
+        dev.apply("CNOT", wires=[0, 1], par=[])
+        dev.apply("CNOT", wires=[1, 2], par=[])
+
+        dev._obs_queue = [
+            qml.PauliZ(wires=[0], do_queue=False)
+            @ qml.Hadamard(wires=[1], do_queue=False)
+            @ qml.PauliY(wires=[2], do_queue=False)
+        ]
+
+        for idx in range(len(dev._obs_queue)):
+            dev._obs_queue[idx].return_type = qml.operation.Sample
+
+        res = dev.pre_measure()
+
+        s1 = dev.sample(["PauliZ", "Hadamard", "PauliY"], [[0], [1], [2]], [[], [], []])
+
+        # s1 should only contain 1 and -1
+        assert np.allclose(s1 ** 2, 1, **tol)
+
+        mean = np.mean(s1)
+        expected = -(np.cos(varphi) * np.sin(phi) + np.sin(varphi) * np.cos(theta)) / np.sqrt(2)
+        assert np.allclose(mean, expected, **tol)
+
+        var = np.var(s1)
+        expected = (
+            3
+            + np.cos(2 * phi) * np.cos(varphi) ** 2
+            - np.cos(2 * theta) * np.sin(varphi) ** 2
+            - 2 * np.cos(theta) * np.sin(phi) * np.sin(2 * varphi)
+        ) / 4
+        assert np.allclose(var, expected, **tol)
+
+    def test_hermitian(self, device, shots, tol):
+        """Test that a tensor product involving qml.Hermitian works correctly"""
+        theta = 0.432
+        phi = 0.123
+        varphi = -0.543
+
+        dev = device(3)
+        dev.apply("RX", wires=[0], par=[theta])
+        dev.apply("RX", wires=[1], par=[phi])
+        dev.apply("RX", wires=[2], par=[varphi])
+        dev.apply("CNOT", wires=[0, 1], par=[])
+        dev.apply("CNOT", wires=[1, 2], par=[])
+
+        A = np.array(
+            [
+                [-6, 2 + 1j, -3, -5 + 2j],
+                [2 - 1j, 0, 2 - 1j, -5 + 4j],
+                [-3, 2 + 1j, 0, -4 + 3j],
+                [-5 - 2j, -5 - 4j, -4 - 3j, -6],
+            ]
+        )
+
+        dev._obs_queue = [
+            qml.PauliZ(wires=[0], do_queue=False) @ qml.Hermitian(A, wires=[1, 2], do_queue=False)
+        ]
+
+        for idx in range(len(dev._obs_queue)):
+            dev._obs_queue[idx].return_type = qml.operation.Sample
+
+        res = dev.pre_measure()
+
+        s1 = dev.sample(["PauliZ", "Hermitian"], [[0], [1, 2]], [[], [A]])
+
+        # s1 should only contain the eigenvalues of
+        # the hermitian matrix tensor product Z
+        Z = np.diag([1, -1])
+        eigvals = np.linalg.eigvalsh(np.kron(Z, A))
+        assert np.allclose(sorted(list(set(s1))), sorted(eigvals), **tol)
+
+        mean = np.mean(s1)
+        expected = 0.5 * (
+            -6 * np.cos(theta) * (np.cos(varphi) + 1)
+            - 2 * np.sin(varphi) * (np.cos(theta) + np.sin(phi) - 2 * np.cos(phi))
+            + 3 * np.cos(varphi) * np.sin(phi)
+            + np.sin(phi)
+        )
+        assert np.allclose(mean, expected, **tol)
+
+        var = np.var(s1)
+        expected = (
```

Comment:
I would just hard-code the expected value here, having it calculated in-place adds no value because you did not parameterize the angles. This also relates to my above comment.

Reply:
This applies also to the other tests.

Reply:
I agree that it makes no difference re: hardcoding the values. At the same time, I would prefer to leave in the full expression for future use, i.e., in case the test does get parametrized, if the precision needs to be adjusted. Also it avoids the expected result looking a bit like a 'magic number'.